include/psa/crypto_builtin_hash.h

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Context structure declaration of the software-based driver which performs
+ *  hashing through the PSA Crypto driver dispatch layer.
```

Comment:
Why a separate header file for each type of algorithm? This seems needlessly complex to me, and not necessarily generalizable to other types of algorithms (e.g. RSA-related definitions may be duplicated between asymmetric encryption and signature, ECC-related definitions between key agreement and signature, block cipher between cipher, AEAD and MAC).